### PR TITLE
Setup docs

### PR DESCRIPTION
**Pull request by 3ygun**
# Goal

Redo the README content to improve understanding for a new user.

# Parts

- [x] Fix a lot of the formatting problems across the README's: headers, syntax highlighting, indentation, etc.
- [x] Add a Dockerfile for environment independent server setup
- [x] Confirm all changes work with @tylermzeller as my Android app wasn't working correctly

# Remaining

Once I can talk to Tyler This should be ready to merge. Please feel free to review the changes. The largest set of changes are to the [firebase/readme.md](firebase/readme.md) so at least look at that one 🌵

### CODE REVIEW

**Comment by tylermzeller** [comment]
We should confirm with @jihunhamm that Docker is a good addition to this project. I have no problems working with it, but I was told that the code base should be as simple as possible.

**Comment by tylermzeller** [comment]
AHH! I just remembered that Docker and Android don't work well together. That is, the Android emulator cannot run while Docker is also running. I ran into this error while trying to do work after I had installed Docker for our capstone project.

**Comment by 3ygun** [comment]
@tylermzeller I added a warning to notate that issue to the reader in `server/readme.md` see 16047b2

File: server/readme.md [hunk 1]
```diff
@@ -1,33 +1,59 @@
...
-7. Run command
+3. The local environment is now setup proceed to **step 4**
 
-  ```
-  $nodejs crowdML-server.js [Constants file name you wish to use]
-  ```
+## Pre-Setup -- Docker
 
-#### What the server-side app does
+**NOTE**: One may not be able to run both Docker and an Android/iOS emulator. Consider using on device builds with the Docker container.
```
Comment (tylermzeller) [inline server/readme.md]: Should we mark this as optional then?

Comment (3ygun) [inline server/readme.md]: I tried to show that with the headers we can talk in class. See the rendered view https://github.com/3ygun/Crowd-ML/tree/setup-docs/server

**Comment by 3ygun** [comment]
All desired parts passing. @jihunhamm could you comment on if you want the `Dockerfile` to run the server?

`Dockerfile` reasoning from @3ygun: Have an isolated [docker](https://www.docker.com/) container for the server to execute within. Implemented as [Dockerfile](https://github.com/3ygun/Crowd-ML/blob/setup-docs/server/Dockerfile) and updated [server README](https://github.com/3ygun/Crowd-ML/tree/setup-docs/server) with setup instructions. 

@tylermzeller reviewed the `README` changes ([see branch](https://github.com/3ygun/Crowd-ML/tree/setup-docs)) for most project `README`'s in `server/`, `firebase/`, `client/Android/`, etc.

**Pull request merged**